PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:4 + add:0, change matches:
-    _inherit = ["product.category"]
-    _inherit = ['res.partner']
-    _inherit = ['sale.order']
-    _inherit = ['sale.order.line']

Total Changes: 10

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/delivery/models/delivery_carrier.py
+++ b/addons/delivery/models/delivery_carrier.py
@@ -11,6 +11,7 @@ from odoo.tools.safe_eval import safe_eval
 
 
 class DeliveryCarrier(models.Model):
+    _name = 'delivery.carrier'
     _description = "Shipping Methods"
     _order = 'sequence, id'
 

--- a/addons/delivery/models/delivery_price_rule.py
+++ b/addons/delivery/models/delivery_price_rule.py
@@ -14,6 +14,7 @@ VARIABLE_SELECTION = [
 
 
 class DeliveryPriceRule(models.Model):
+    _name = 'delivery.price.rule'
     _description = "Delivery Price Rules"
     _order = 'sequence, list_price, id'
 

--- a/addons/delivery/models/product_category.py
+++ b/addons/delivery/models/product_category.py
@@ -5,7 +5,7 @@ from odoo.exceptions import UserError
 
 
 class ProductCategory(models.Model):
-    _inherit = ["product.category"]
+    _inherit = "product.category"
 
     @api.ondelete(at_uninstall=False)
     def _unlink_except_delivery_category(self):

--- a/addons/delivery/models/res_partner.py
+++ b/addons/delivery/models/res_partner.py
@@ -4,6 +4,6 @@ from odoo import fields, models
 
 
 class ResPartner(models.Model):
-    _inherit = ['res.partner']
+    _inherit = 'res.partner'
 
     property_delivery_carrier_id = fields.Many2one('delivery.carrier', company_dependent=True, string="Delivery Method", help="Default delivery method used in sales orders.")

--- a/addons/delivery/models/sale_order.py
+++ b/addons/delivery/models/sale_order.py
@@ -7,7 +7,7 @@ from odoo.exceptions import UserError
 
 
 class SaleOrder(models.Model):
-    _inherit = ['sale.order']
+    _inherit = 'sale.order'
 
     pickup_location_data = fields.Json()
     carrier_id = fields.Many2one('delivery.carrier', string="Delivery Method", check_company=True, help="Fill this field if you plan to invoice the shipping based on picking.")

--- a/addons/delivery/models/sale_order_line.py
+++ b/addons/delivery/models/sale_order_line.py
@@ -4,7 +4,7 @@ from odoo import api, fields, models
 
 
 class SaleOrderLine(models.Model):
-    _inherit = ['sale.order.line']
+    _inherit = 'sale.order.line'
 
     is_delivery = fields.Boolean(string="Is a Delivery", default=False)
     product_qty = fields.Float(
